### PR TITLE
tiny diffsync followups

### PR DESCRIPTION
**Pull request by phritz**
progress towards https://github.com/rocicorp/repc/issues/290

### CODE REVIEW

File: src/sync/pull.rs [hunk 1]
```diff
@@ -281,6 +283,21 @@ pub async fn maybe_end_try_pull(
         .await
         .map_err(WriteSyncHeadError)?;
     dag_write.commit().await.map_err(CommitError)?;
+    let (old_last_mutation_id, old_cookie) = Commit::snapshot_meta_parts(&main_snapshot)
```
Comment (arv) [inline src/sync/pull.rs:286]: Could these be wrapped in a `debug` check? Seems silly to do this work if we are not going to log them (which we do not do by default)

**Review by arv — COMMENTED** [review]
LGTM

I always imagined we would include the `app_version`/`schemaVersion` in `push` too. It seems important for the server to know what version the client is at in a push request too.

**Comment by phritz** [comment]
> I always imagined we would include the app_version/schemaVersion in push too. It seems important for the server to know what version the client is at in a push request too.

Recall for pull that pull_version indicates the version of the protocol the client is using which is hard-coded in repc, and schema_version indicates what kind of client view the app understands and is set by the customer's code. For push, we already have push_version to indicate the version of the protocol repc is using. I think what you are asking is that we include a schema_version in push as well, that would indicate... what? The schema of the mutator args? That makes sense I guess but want to be sure what you are suggesting.

**Comment by arv** [comment]
I have been arguing for DO_FOO_v2 in the past. Lets just stick with that story until costumers keep shooting themselves in the foot.

> I think what you are asking is that we include a schema_version in push as well, that would indicate... what? The schema of the mutator args? 

Correct

**Comment by phritz** [comment]
Well, too late :) https://github.com/rocicorp/repc/pull/327

File: src/sync/pull.rs (hunk 1 above)
Comment (phritz) [reply src/sync/pull.rs:286]: yes, done in https://github.com/rocicorp/repc/pull/327/commits/46d67d234a322306e94c2da01711b7099c56e3a6

**Pull request merged**